fv3gfs-util/tests/test_partitioner.py

```diff
@@ -108,150 +107,436 @@ def test_tile_extent_from_rank_metadata(array_extent, array_dims, layout, tile_e
+            (3, 3),
+            8,
+            (slice(0, 5), slice(7, 8), slice(7, 8)),
+            False,
+            float(1 / 6),
+            id="54_rank_topright_right_no_overlap_square_layout_half_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 16, 16),
+            (4, 4),
+            0,
+            (slice(0, 5), slice(0, 2), slice(0, 2)),
+            False,
+            float(1 / 3),
+            id="96_rank_farbotfarleft_right_no_overlap_square_layout_third_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 16, 16),
+            (4, 4),
+            1,
+            (slice(0, 5), slice(0, 2), slice(2, 8)),
+            False,
+            float(1 / 3),
+            id="96_rank_farbotcloseleft_right_no_overlap_square_layout_third_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 16, 16),
+            (4, 4),
+            6,
+            (slice(0, 5), slice(2, 8), slice(8, 14)),
+            False,
+            float(1 / 3),
+            id="96_rank_closebotcloseright_right_no_overlap_square_layout_third_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 16, 16),
+            (4, 4),
+            14,
+            (slice(0, 5), slice(14, 16), slice(8, 14)),
+            False,
+            float(1 / 3),
+            id="96_rank_fartopcloseright_right_no_overlap_square_layout_third_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Y_INTERFACE_DIM, fv3gfs.util.X_INTERFACE_DIM],
+            (13, 13),
+            (2, 4),
+            1,
+            (slice(0, 7), slice(3, 7)),
+            True,
+            1.0,
+            id="48_rank_botcloseleft_interface_right_overlap",
+        ),
+        pytest.param(
+            [fv3gfs.util.Y_INTERFACE_DIM, fv3gfs.util.X_INTERFACE_DIM],
+            (13, 13),
+            (2, 4),
+            1,
+            (slice(0, 7), slice(2, 7)),
+            True,
+            0.5,
+            id="48_rank_botcloseleft_interface_right_overlap_half_edge",
+        ),
+        pytest.param(
+            [fv3gfs.util.Y_INTERFACE_DIM, fv3gfs.util.X_INTERFACE_DIM],
+            (13, 13),
+            (2, 4),
+            7,
+            (slice(6, 13), slice(10, 13)),
+            True,
+            0.5,
+            id="48_rank_topfarright_interface_right_overlap_half_edge",
+        ),
+        pytest.param(
+            [fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (12, 12),
+            (2, 4),
+            0,
+            (slice(0, 6), slice(0, 2)),
+            True,
+            0.5,
+            id="48_rank_botfarleft_right_overlap_half_edge",
+        ),
```

Comment:
If you can, please add a test case or an entirely separate test for layout=(5, 5), array size = (16, 16), dims=[X_DIM, Y_DIM], tile_ratio=1. The "correct" decomposition from Fortran has length per rank of (3, 3, 4, 3, 3). The current code does not support this case - you can make a test that simply asserts an exception should be raised (a ValueError) by using
```
with pytest.raises(ValidationError):
    function_that_raises_a_validation_error()
```
That line will fail if the expected exception is not raised and continue if it is raised.

Reply:
The code now supports this case, it will find an valid assignment of (2,4,4,4,2). 
I have however added tests for a case where the grid has an odd number of points along an axis and the number of ranks along the same axis is even.